projects/pastanaga-angular/src/lib/controls/form-field/pa-form-control.directive.ts

```diff
@@ -0,0 +1,278 @@
+import {
+    ChangeDetectorRef,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    Self,
+    SimpleChanges,
+} from '@angular/core';
+import {
+    ControlValueAccessor,
+    FormControl,
+    FormControlDirective,
+    FormControlName,
+    NgControl,
+    NgModel,
+    ValidatorFn,
+} from '@angular/forms';
+import { FORM_CONTROL, FORM_CONTROL_NAME, InternalMode, NG_MODEL, STANDALONE } from '../form-field.model';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { markForCheck } from '../../common';
+import { buildAlwaysFalseValidator } from '../form-field.utils';
+import { Subject } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+
+let nextId = 0;
+
+@Directive({
+    selector: '[paFormControl]',
+})
+export class PaFormControlDirective implements OnChanges, OnInit, OnDestroy, ControlValueAccessor {
+    @Input() set id(value: string | undefined) {
```

Comment:
It could be useful know if the form control is 'pristine'.

Reply:
we can easily provide more information in our statusChange emitter... to be discussed